tests/passes/compile-sync/compile-sync.expect

```diff
@@ -0,0 +1,222 @@
+      addr.write_en = 1'd1;
+      add_1.right = 3'd1;
+      add_1.left = addr.out;
+      addr.in = add_1.out;
+      incr_idx[done] = addr.done;
+    }
+    group wait_restore {
+      wait_restore_reg.write_en = !eq.out ? 1'd1;
+      wait_restore_reg.in = !eq.out ? 1'd1;
+      wait_restore[done] = wait_restore_reg.done;
+    }
+    group restore {
+      barrier.in_0 = 32'd0;
+      barrier.write_en_0 = 1'd1;
+      restore[done] = barrier.write_done_0;
+    }
+    group clear_barrier {
+      barrier.read_en_0 = 1'd1;
+      clear_barrier[done] = barrier.read_done_0;
+    }
+    group incr_barrier {
+      barrier.read_en_0 = 1'd1;
+      save.write_en = barrier.read_done_0;
+      incr.right = barrier.read_done_0 ? 32'd1;
+      incr.left = barrier.read_done_0 ? barrier.out_0;
+      save.in = barrier.read_done_0 ? incr.out;
+      incr_barrier[done] = save.done;
+    }
+    group write_barrier {
+      barrier.in_0 = save.out;
+      barrier.write_en_0 = 1'd1;
+      write_barrier[done] = barrier.write_done_0;
+    }
+    group wait {
+      wait_reg.write_en = eq.out ? 1'd1;
+      wait_reg.in = eq.out;
+      wait[done] = wait_reg.done;
+    }
+    group wait_restore0 {
+      wait_restore_reg0.write_en = !eq0.out ? 1'd1;
+      wait_restore_reg0.in = !eq0.out ? 1'd1;
+      wait_restore0[done] = wait_restore_reg0.done;
+    }
+    group restore0 {
+      barrier0.in_0 = 32'd0;
+      barrier0.write_en_0 = 1'd1;
+      restore0[done] = barrier0.write_done_0;
+    }
+    group clear_barrier0 {
+      barrier0.read_en_0 = 1'd1;
+      clear_barrier0[done] = barrier0.read_done_0;
+    }
+    group incr_barrier0 {
+      barrier0.read_en_0 = 1'd1;
+      save0.write_en = barrier0.read_done_0;
+      incr0.right = barrier0.read_done_0 ? 32'd1;
+      incr0.left = barrier0.read_done_0 ? barrier0.out_0;
+      save0.in = barrier0.read_done_0 ? incr0.out;
+      incr_barrier0[done] = save0.done;
+    }
+    group write_barrier0 {
+      barrier0.in_0 = save0.out;
+      barrier0.write_en_0 = 1'd1;
+      write_barrier0[done] = barrier0.write_done_0;
+    }
+    group wait0 {
+      wait_reg0.write_en = eq0.out ? 1'd1;
+      wait_reg0.in = eq0.out;
+      wait0[done] = wait_reg0.done;
+    }
+    group incr_barrier1 {
+      barrier.read_en_1 = 1'd1;
+      save1.write_en = barrier.read_done_1;
+      incr1.right = barrier.read_done_1 ? 32'd1;
+      incr1.left = barrier.read_done_1 ? barrier.out_1;
+      save1.in = barrier.read_done_1 ? incr1.out;
+      incr_barrier1[done] = save1.done;
+    }
+    group write_barrier1 {
+      barrier.in_1 = save1.out;
+      barrier.write_en_1 = 1'd1;
+      write_barrier1[done] = barrier.write_done_1;
+    }
+    group wait1 {
+      wait_reg1.write_en = eq.out ? 1'd1;
+      wait_reg1.in = eq.out;
+      wait1[done] = wait_reg1.done;
+    }
+    group incr_barrier2 {
+      barrier0.read_en_1 = 1'd1;
+      save2.write_en = barrier0.read_done_1;
+      incr2.right = barrier0.read_done_1 ? 32'd1;
```

Comment:
Is there a reason to guard the writes to the `incr.left` and `incr.right` ports with the `read_done_1` signal? We can just continuously assign to them because we only read `incr.out` when `read_done_1` is assserted